src/workflows-app/SubmissionDetails.test.js

```diff
@@ -690,4 +779,310 @@ describe('Submission Details page', () => {
   });
+
+  it('should open the log viewer modal when Execution Logs button is clicked', async () => {
+    // Arrange
+    const user = userEvent.setup();
+    const getRuns = jest.fn(() => Promise.resolve(mockRunsData));
+    const getRunsSets = jest.fn(() => Promise.resolve(runSetData));
+    const getMethods = jest.fn(() => Promise.resolve(methodData));
+    const mockLeoResponse = jest.fn(() => Promise.resolve(mockAzureApps));
+    Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runs: {
+            get: getRuns,
+          },
+          runSets: {
+            get: getRunsSets,
+          },
+          methods: {
+            getById: getMethods,
+          },
+        },
+        Apps: {
+          listAppsV2: mockLeoResponse,
+        },
+        CromwellApp: {
+          workflows: () => {
+            return {
+              metadata: jest.fn(() => {
+                return Promise.resolve(runDetailsMetadata);
+              }),
+            };
+          },
+        },
+        AzureStorage: {
+          blobByUri: jest.fn(() => ({
+            getMetadataAndTextContent: () =>
+              Promise.resolve({
+                uri: 'https://someBlobFilePath.blob.core.windows.net/cromwell/user-inputs/inputFile.txt',
+                sasToken: '1234-this-is-a-mock-sas-token-5678',
+                fileName: 'inputFile.txt',
+                name: 'inputFile.txt',
+                lastModified: 'Mon, 22 May 2023 17:12:58 GMT',
+                size: '324',
+                contentType: 'text/plain',
+                textContent: 'this is the text of a mock file',
+                azureSasStorageUrl: 'https://someBlobFilePath.blob.core.windows.net/cromwell/user-inputs/inputFile.txt',
+              }),
+          })),
+          details: jest.fn().mockResolvedValue({ sas: { token: '1234-this-is-a-mock-sas-token-5678' } }),
+        },
+        Metrics: {
+          captureEvent,
+        },
+      };
+    });
+
+    // Act
+    await act(async () => {
+      render(
+        h(BaseSubmissionDetails, {
+          name: 'test-azure-ws-name',
+          namespace: 'test-azure-ws-namespace',
+          workspace: mockAzureWorkspace,
+          submissionId,
+          workflowId: '00001111-2222-3333-aaaa-bbbbccccdddd',
+          uri: 'https://coaexternalstorage.blob.core.windows.net/cromwell/user-inputs/inputFile.txt',
+        })
+      );
+    });
+
+    const executionLogsButtons = screen.getAllByRole('button', { name: 'Logs' });
+
+    for (const executionLogsButton of executionLogsButtons) {
+      // Act
+      await user.click(executionLogsButton);
+
+      // Assert
+      screen.getByRole('dialog', { name: 'Workflow Execution Log' });
+      screen.getByText('File:');
+      screen.getByText('workflow.log');
+      screen.findByRole('link', { name: 'Download log' });
+      screen.getByRole('button', { name: 'Workflow Execution Log' });
+      screen.findByText('this is the text of a mock file');
```

Comment:
`findBy...` queries return a Promise and thus would need to be awaited. Can these be replaced with `getBy...` queries?
https://testing-library.com/docs/queries/about#types-of-queries

```suggestion
      screen.getByRole('link', { name: 'Download log' });
      screen.getByRole('button', { name: 'Workflow Execution Log' });
      screen.getByText('this is the text of a mock file');
```